keymanapp/keyman
# feat(windows): add right modifier included in hotkey optional functionality

Step 2. of Fixes:#11471

# User Testing

## TEST_LANGUAGE_HOTKEYS_LEFT_SIDE

For this test only the left and right modifier keys will trigger the hotkeys
1. Install Keyman
2. Install some keyboard layouts Tamil and IPA SIL
3. Open Configuration and assign hotkeys to <kbd>Ctrl</kbd>+<kbd>G</kbd> to Tamil and 
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd>to IPA SIL
5. Open Notepad and type some text 
6. Switch to Tamil using the Hotkey press <kbd>Left Ctrl</kbd>+<kbd>G</kbd>
7. Verify that it has switched in the task and language bar to Tamil and type some text.
8. Switch to IPA SIL using the Hotkey press <kbd>Left Ctrl</kbd>+<kbd>Left Shift</kbd>+<kbd>I</kbd>
9. Verify it has it has switched in the task and language bar to IPA SIL and type some text.

## TEST_LANGUAGE_HOTKEYS_RIGHT_SIDE

For this test only the left and right modifier keys will trigger the hotkeys
1. Install Keyman
2. Install some keyboard layouts Tamil and IPA SIL
3. Open Configuration and assign hotkeys to <kbd>Ctrl</kbd>+<kbd>G</kbd> to Tamil and 
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd>to IPA SIL
4. In Configuration Options tab click the check box for "Use right modifier for hotkeys"
5. Open Notepad and type some text 
6. Switch to Tamil using the Hotkey  press <kbd>Right Ctrl</kbd>+<kbd>G</kbd>
7. Verify that it has switched in the task and language bar to Tamil and type some text.
8. Switch to IPA SIL using the Hotkey  press <kbd>Ctrl</kbd>+<kbd>Right Shift</kbd>+<kbd>I</kbd>
9. Verify it has it has switched in the task and language bar to IPA SIL and type some text.
